pkg/signal/testfiles/main.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"os"
+	"syscall"
+	"time"
+
+	"github.com/docker/docker/pkg/signal"
+)
+
+func main() {
+	sigmap := map[string]os.Signal{
+		"TERM": syscall.SIGTERM,
+		"QUIT": syscall.SIGQUIT,
+		"INT":  os.Interrupt,
+	}
+
+	defer time.Sleep(10 * time.Millisecond)
+
+	signal.Trap(func() {
+		time.Sleep(10 * time.Millisecond)
```

Comment:
This defer is written because we want the process to actually wait for the signals that we are providing to the process. This time doesn't actually complete because we terminate our process by providing it `SIGTERM` or `SIGKILL`. We want to observe if the cleanup function is being called correctly or not.
So, if all goes correctly this wait should not complete.